ENIGMAsystem/SHELL/Universal_System/Resources/fontstruct.cpp

```diff
@@ -36,35 +36,25 @@ using std::string;
 
 namespace enigma
 {
-  font **fontstructarray = NULL;
-  extern size_t font_idmax;
+  AssetArray<font> fonts;
 
   bool fontglyph::empty() {
     return !(std::abs(x2-x) > 0 && std::abs(y2-y) > 0);
   }
 
   int font_new(uint32_t gs, uint32_t gc) // Creates a new font, allocating 'gc' glyphs
   {
-    font *ret = new font;
+    font ret;
 
     fontglyphrange fgr;
     fgr.glyphstart = gs;
     fgr.glyphcount = gc;
 
-    ret->glyphRangeCount = 1;
-    ret->glyphRanges.push_back(fgr);
-    ret->height = 0;
+    ret.glyphRangeCount = 1;
```

Comment:
I wanted to remove this variable as we should just be able to callfont.fgrs.size() insread of using this variable.

Reply:
Ok: a970f47c56390459291794e1bfcd0ee4951bb484

That was an easy refactor. The counts being in the runtime fonts was actually just vestigial, I didn't need to do that when I originally added this. The counts are only needed for the raw fonts to be read off the exe.

Josh has expressed that this should be moved to a code generated header, rather than appending this meta information to the game executable. That would make debugging games easier to inspect.